src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -138,13 +142,15 @@ public class RSessionProvider : IRSessionProvider {
                         foreach (var session in sessions) {
                             session.CancelSwitchingBroker();
                         }
+                        BrokerChangeFailed?.Invoke(this, EventArgs.Empty);
```

Comment:
I know - but we can add handler to display UI for example. I just didn't want to fire 'Changing' and then nothing if it fails.